### ISSUE TITLE
Not support custom field options 

### ISSUE THREAD

**Issue by hiyoi** [nipunn1313/mypy-protobuf#363]
here is my example file `abcd.proto`
```
syntax = "proto3";

import "google/protobuf/descriptor.proto";

package abcd;

message AbcdMessage {
  string id = 1 [(myfield).column="id"];
}

message MyFieldOptions {
    string column = 1;
}

extend google.protobuf.FieldOptions {
    MyFieldOptions myfield = 60000;
}
```

run
`
protoc -I=path/to/src --python_out=path/to/out  path/to/src/abcd.proto
`
work fine

run 
`protoc -I=path/to/src --python_out=path/to/out  --mypy_out=path/to/out path/to/src/abcd.proto`

got error
`
AssertionError: mypy_protobuf.[casttype,keytype,valuetype] is expected to be of format path/to/file.TypeInFile
--mypy_out: protoc-gen-mypy: Plugin failed with status code 1.
`

reference: 
https://developers.google.com/protocol-buffers/docs/proto#customoptions

**Comment by nipunn1313** [comment]
Thanks for the bugreport! Seems like a bug! At the least we shouldn't crash here.

The error message doesn't really make any sense either.

**Comment by hiyoi** [comment]
> Thanks for the bugreport! Seems like a bug! At the least we shouldn't crash here.
> 
> The error message doesn't really make any sense either.

have any solution?

**Comment by nipunn1313** [comment]
Haven't worked on it yet! PR's welcome if it's urgent for you.

Otherwise, I'll try to find some time to work on it.

**Comment by nipunn1313** [comment]
I actually poked around and custom field options are supported.
The issue arises because (currently) mypy-protobuf uses extension numbers 60000-60004 for internal purposes.

The correct way to solve this is by reserving some globally unique numbers within the space here https://chromium.googlesource.com/chromium/src/+/master/third_party/protobuf/docs/options.md
I will work on that.

For now, you can workaround this issue by using a different extension number.

**Comment by nipunn1313** [comment]
Filed #396 
will close this

https://github.com/nipunn1313/mypy-protobuf/blob/main/proto/mypy_protobuf/extensions.proto
for context on the conflict

**Comment by browncrane** [comment]
With version 3.3.0 the bug still exists.
```
pip list | grep mypy
mypy               0.950
mypy-extensions    0.4.3
mypy-protobuf      3.3.0
```

**Comment by nipunn1313** [comment]
Hi. The fix #403 is not yet released.
Will be working on an upcoming release with the latest changes when I get some time.

As mentioned earlier, there is a workaround:

> For now, you can workaround this issue by using a different extension number.
